packages/core/test/node/com.spec.ts

```diff
@@ -95,4 +97,36 @@ describe('Communication', () => {
 
         expect(res).to.be.equal('child echo Yoo!');
     });
+
+    it('doesnt send callback message on a method that was defined not to send one', async () => {
+        const host = new BaseHost();
+        const main = new Communication(host, 'main', undefined, undefined, undefined, {
+            warnOnSlow: true
+        });
+
+        const host2 = host.open();
+        const child = new Communication(host2, 'child', undefined, undefined, undefined, {
+            warnOnSlow: true
+        });
+
+        const handleMessageStub = stub(main, 'handleMessage');
+        const childCallMethodStub = stub(child, 'callMethod');
+
+        main.registerEnv('child', host2);
+
+        child.registerAPI({ id: 'echoService' }, new EchoService());
+        const proxy = main.apiProxy<EchoService>(
+            Promise.resolve({ id: 'child' }),
+            { id: 'echoService' },
+            {
+                echo: {
+                    emitOnly: true
+                }
+            }
+        );
+        await proxy.echo('Yo!');
+
+        expect(childCallMethodStub).to.have.not.been.called;
```

Comment:
something here needs to be checked it should call callMethod on the child

Reply:
Written an explanation